Fix Swift 5.7 warnings.

robo-fish opened the pull request:
The following warning is emitted for a number of function declarations when compiling with Swift 5.7 (Xcode 14 Beta):
*"Non-'@objc' instance method in extensions cannot be overridden; use 'public' instead"*

Changing the access levels of the functions from *open* to *public* resolves the warnings.

robo-fish committed:
Fix Swift 5.7 warnings.

"Non-'@objc' instance method in extensions cannot be overridden; use 'public' instead"

jormungand2006 commented:
Hi @dehesa, 
Apple has already released up to Xcode 14.2, do you plan to merge this pull request with the latest release?

robo-fish committed:
Fix Swift 5.8 warning.

'Foundation' was not imported by this file.

inc-ali commented:
@dehesa please accept this pull request :)

robo-fish committed:
Work around Swift 5.10 compiler bug.

The workaround fixes an internal Swift 5.10 compiler bug,
where an error is thrown when 'NSDecimalString' is encountered
in the source code.

As a side effect, Swift tools version is upgraded to 5.9,
and the deployment platforms are drastically narrowed down.

robo-fish committed:
Fix compatibility with Swift 6 language mode.
